partiql-lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt

```diff
@@ -79,7 +79,7 @@ interface CompilerPipeline {
      * Includes built-in functions as well as custom functions added while the [CompilerPipeline]
      * was being built.
      */
-    val functions: @JvmSuppressWildcards Map<String, ExprFunction>
```

Comment:
We'll need to provide an alternative to the `functions` property (maybe `functionList`) and deprecate the old property.

Reply:
I don't think we need to deprecate this because the `data class StepContext` is only invoked inside `CompilerPipelineImpl` where we already add deprecate notice to functionMap. In this way,  `CompilerPipelineImpl` always convert functionMap to a list so StepContext will just use list.

Reply:
I don't think we need to deprecate this because in `CompilerPipelineImpl`, I already add deprecate notice to the old functionMap. In this way, `CompilerPipelineImpl` always convert functionMap to a list so `CompilerPipeline` interface don't need to have the map representation.